engine/src/main/java/io/zeebe/engine/processing/bpmn/BpmnElementContainerProcessor.java

```diff
@@ -35,14 +35,27 @@ void onChildActivating(
       final BpmnElementContext childContext);
 
   /**
-   * A child element is completed. Leave the element container if it has no more active child
-   * elements.
+   * The execution path of a child element is about to be completed.
    *
    * @param element the instance of the BPMN element container
    * @param flowScopeContext process instance-related data of the element container
-   * @param childContext process instance-related data of the child element that is completed
+   * @param childContext process instance-related data of the child element that is completed. At
+   *     this point in time the element is still present in the state
    */
-  void onChildCompleted(
+  void beforeExecutionPathCompleting(
```

Comment:
I understand the split into before and after completing, however we already have concepts for this: Completing and Completed. Completing is the state of the element when it starts to complete, while completed is the state of the element when it has completed. I think we can use `onExecutionPathCompleting` and `onExecutionPathCompleted` respectively.

Reply:
Naming discussed f2f